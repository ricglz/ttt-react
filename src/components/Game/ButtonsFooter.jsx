import React from 'react';
import PropTypes from 'prop-types';
import BackButton from '../Layout/BackButton';
import ResetButton from '../Layout/ResetButton';

const ButtonsFooter = ({ reset, back }) => (
  <div className="row justify-content-center">
    <ResetButton onClick={reset} />
    <BackButton text="shared.back" url={back} />
  </div>
);

ButtonsFooter.propTypes = {
  reset: PropTypes.func.isRequired,
  back: PropTypes.string.isRequired,
};

export default ButtonsFooter;
