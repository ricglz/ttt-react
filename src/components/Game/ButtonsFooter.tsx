import React from 'react';
import BackButton from '../Layout/BackButton';
import ResetButton from '../Layout/ResetButton';

type Props = {
  reset: () => void,
  back: string,
};

const ButtonsFooter = ({ reset, back }: Props) => (
  <div className="row justify-content-center">
    <ResetButton onClick={reset} />
    <BackButton text="shared.back" url={back} />
  </div>
);

export default ButtonsFooter;
