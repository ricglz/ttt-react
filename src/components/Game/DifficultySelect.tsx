import Select from 'react-select';
import type { ActionMeta, SingleValue } from 'react-select';

import type { Difficulty } from '../../@types/general_enums';
import { FormattedMessage } from '../Layout/FormattedText';

export type Option = {
  label: object;
  value: Difficulty;
};

const easy = <FormattedMessage locale="game.easy" />;
const medium = <FormattedMessage locale="game.medium" />;
const hard = <FormattedMessage locale="game.hard" />;
const options: Option[] = [
  { value: 1, label: easy },
  { value: 2, label: medium },
  { value: 3, label: hard },
];
const placeholder = <FormattedMessage locale="game.placeholder" />;

type Props = {
  selectedOption: Option | null;
  handleChange: (
    newValue: SingleValue<Option>,
    actionMeta: ActionMeta<Option>
  ) => void;
};

const DifficultySelect = ({ selectedOption, handleChange }: Props) => (
  <div className="row justify-content-center mb-3">
    <div className="col-12 col-sm-10 col-md-8 col-lg-6 col-xl-5">
      <Select
        placeholder={placeholder}
        value={selectedOption}
        onChange={handleChange}
        options={options}
      />
    </div>
  </div>
);

export default DifficultySelect;
