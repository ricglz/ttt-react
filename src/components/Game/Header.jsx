import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { FormattedHeader, FormattedHeader2 } from '../Layout/FormattedText';

const Score = ({ score, klass, defaultMessage }) => (
  <div className={`col ${klass}`}>
    <p>
      <FormattedMessage id={`game.${klass}`} default={defaultMessage} />
      {` ${score}`}
    </p>
  </div>
);

const ScoresSection = ({ xScore, oScore }) => (
  <div className="col-12">
    <div className="row justify-content-between">
      <Score score={xScore} klass="x-score" defaultMessage="X's score" />
      <Score score={oScore} klass="o-score" defaultMessage="O's score" />
    </div>
  </div>
);

const Header = ({ xScore, oScore, ai }) => (
  <div className="row">
    <div className="col-12">
      <FormattedHeader locale={ai ? 'shared.sp' : 'shared.mp'} default="Type of game" />
    </div>
    <div className="col-12">
      <FormattedHeader2 locale="game.score" default="Score" />
    </div>
    <ScoresSection xScore={xScore} oScore={oScore} />
  </div>
);

Score.propTypes = {
  score: PropTypes.number.isRequired,
  klass: PropTypes.string.isRequired,
  defaultMessage: PropTypes.string.isRequired,
};

ScoresSection.propTypes = {
  oScore: PropTypes.number.isRequired,
  xScore: PropTypes.number.isRequired,
};

Header.propTypes = {
  ai: PropTypes.bool.isRequired,
  oScore: PropTypes.number.isRequired,
  xScore: PropTypes.number.isRequired,
};

export default Header;
