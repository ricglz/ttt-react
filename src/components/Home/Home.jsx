import React from 'react';
import { FormattedMessage } from 'react-intl';
import HomeButton from './HomeButton';

function Home() {
  return (
    <div className="container text-center">
      <div className="row">
        <div className="col">
          <FormattedMessage id="homePage.title" default="Home Page">
            {txt => <h1>{txt}</h1>}
          </FormattedMessage>
        </div>
      </div>
      <HomeButton text="shared.sp" url="/singleplayer" />
      <HomeButton text="shared.mp" url="/multiplayer" />
      <HomeButton staticText text="Online" url="/online" />
      <HomeButton text="shared.tutorial" url="/tutorial" />
      <HomeButton staticText text="Contributors" url="/contributors" />
    </div>
  );
}

export default Home;
