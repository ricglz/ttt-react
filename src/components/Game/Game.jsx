import React from 'react';
import PropTypes from 'prop-types';
import Header from './Header';
import BigBoard from '../Board/BigBoard';
import ButtonsFooter from './ButtonsFooter';
import DifficultySelect from './DifficultySelect';
import {
  theresAWinner,
  initialState,
  alertWinner,
} from '../../functions/HelperFunctions';
import {
  playerMadeAMove,
} from '../../functions/Ai';
import { useGameHooks, useAIHooks } from '../../functions/CustomHooks';

function Game({ ai }) {
  const [game, setGame] = React.useState(initialState());
  const [oWins, setOWins] = React.useState(0);
  const [xWins, setXWins] = React.useState(0);
  const [selectedOption, setSelectedOption] = React.useState(null);
  const {
    boardGame, currentBoard, moveNumber, currentPlayer,
  } = game;

  const [canClick, pvpMove, newGame] = useGameHooks(game, setGame);

  const changeScore = React.useCallback((value) => {
    if (value === -1) {
      setOWins(oWins + 1);
    } else {
      setXWins(xWins + 1);
    }
  }, [setOWins, setXWins, oWins, xWins]);

  const [aiMove, handleChange] = useAIHooks({
    moveNumber,
    selectedOption,
    setGame,
    game,
    changeScore,
    newGame,
    setSelectedOption,
  });

  const handleSquareClick = React.useCallback((board, id) => {
    if (canClick(board, id)) {
      const boardCopy = [...boardGame];
      const newMoveNumber = moveNumber + 1;
      boardCopy[board][id] = currentPlayer === 'X' ? 1 : -1;
      const winner = theresAWinner(boardCopy[board]);
      if (winner) {
        alertWinner(winner);
        changeScore(winner);
        newGame();
      } else if (newMoveNumber === 81) {
        newGame();
      } else if (ai) {
        playerMadeAMove(board, boardCopy[board]);
        aiMove(boardCopy, id, newMoveNumber);
      } else {
        pvpMove(boardCopy, newMoveNumber, id);
      }
    }
  }, [
    canClick, boardGame, moveNumber, currentPlayer, ai, changeScore, newGame, aiMove, pvpMove,
  ]);

  return (
    <div className="container text-center">
      <Header ai={ai} oScore={oWins} xScore={xWins} />
      <hr />
      <BigBoard
        handleClick={handleSquareClick}
        boardGame={boardGame}
        currentBoard={currentBoard}
      />
      <hr />
      {ai && (
      <DifficultySelect
        selectedOption={selectedOption}
        handleChange={handleChange}
      />
      )}
      <ButtonsFooter back="/" reset={newGame} />
    </div>
  );
}

Game.propTypes = {
  ai: PropTypes.bool,
};

Game.defaultProps = {
  ai: false,
};
export default Game;
