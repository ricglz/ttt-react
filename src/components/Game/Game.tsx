import React from 'react';
import Header from './Header';
import BigBoard from '../Board/BigBoard';
import ButtonsFooter from './ButtonsFooter';
import type { Option } from './DifficultySelect';
import DifficultySelect from './DifficultySelect';
import {
  initialState,
} from '../../functions/HelperFunctions';
import {
  useGameHooks, useAIHooks, useScore, useAfterMove, useHandleClick,
} from '../../functions/GameHooks';
import Ai from '../../functions/Ai';

type Props = {
  isAi?: boolean
};

export default function Game({ isAi = true }: Props) {
  const [game, setGame] = React.useState(initialState());
  const [selectedOption, setSelectedOption] = React.useState<Option | null>(null);
  const { oWins, xWins, changeScore } = useScore();
  const {
    boardGame, currentBoard, moveNumber, currentPlayer,
  } = game;
  const ai = React.useState(
    () => new Ai({ board: boardGame[0], currentBoard: 0, currentDifficulty: 1 }),
  )[0];
  const { canClick, pvpMove, newGame } = useGameHooks(game, setGame, ai);
  const { aiMove, handleChange } = useAIHooks({
    moveNumber,
    selectedOption,
    setGame,
    game,
    changeScore,
    newGame,
    setSelectedOption,
    ai,
  });
  const afterMove = useAfterMove({
    ai, aiMove, changeScore, newGame, pvpMove, isAi,
  });
  const handleSquareClick = useHandleClick({
    canClick, boardGame, moveNumber, currentPlayer, afterMove,
  });

  return (
    <div className="container text-center">
      <Header ai={isAi} oScore={oWins} xScore={xWins} />
      <hr />
      <BigBoard
        handleClick={handleSquareClick}
        boardGame={boardGame}
        currentBoard={currentBoard}
      />
      <hr />
      {ai && (
        <DifficultySelect
          selectedOption={selectedOption}
          handleChange={handleChange}
        />
      )}
      <ButtonsFooter back="/" reset={newGame} />
    </div>
  );
}
