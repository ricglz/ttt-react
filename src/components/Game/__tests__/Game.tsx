import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from '../../Home/Home';
import Game from '../Game';
import ContextsProvider from '../../../ContextsProvider';

function Component() {
  return (
    <ContextsProvider
      ChildrenComponent={Home}
      RouterComponent={MemoryRouter}
    >
      <Game />
    </ContextsProvider>
  );
}

it('renders without crashing', () => {
  const component = <Component />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
