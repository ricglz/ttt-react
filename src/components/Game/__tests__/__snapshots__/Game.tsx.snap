// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="text-left"
  dir="ltr"
>
  <div
    className="container text-center"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h1>
          Home Page
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/singleplayer"
        onClick={[Function]}
      >
        Single Player
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/multiplayer"
        onClick={[Function]}
      >
        Local Multiplayer
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/online"
        onClick={[Function]}
      >
        <span>
          Online
        </span>
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/tutorial"
        onClick={[Function]}
      >
        How to play
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/contributors"
        onClick={[Function]}
      >
        <span>
          Contributors
        </span>
      </a>
    </div>
  </div>
  <div
    className="container text-center"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h1>
          Single Player
        </h1>
      </div>
      <div
        className="col-12"
      >
        <h2>
          Score
        </h2>
      </div>
      <div
        className="col-12"
      >
        <div
          className="row justify-content-between"
        >
          <div
            className="col x-score"
          >
            <p>
              X's score
               0
            </p>
          </div>
          <div
            className="col o-score"
          >
            <p>
              O's score
               0
            </p>
          </div>
        </div>
      </div>
    </div>
    <hr />
    <div
      className="game row justify-content-center"
    >
      <div
        className="col-9 col-sm-8 col-md-7 col-lg-6 col-xl-6"
      >
        <div
          className="game row"
        >
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="game row"
        >
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="game row"
        >
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-4 big-box"
          >
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
            <div
              className="game row cell-row"
            >
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
              <div
                className="col-4 box"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <p
                  className=""
                >
                  
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr />
    <div
      className="row justify-content-center mb-3"
    >
      <div
        className="col-12 col-sm-10 col-md-8 col-lg-6 col-xl-5"
      >
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-1s2u09g-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-14el2xx-placeholder"
                id="react-select-2-placeholder"
              >
                Choose the difficulty (Default easy)
              </div>
              <div
                className=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-2-listbox"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-owns="react-select-2-listbox"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-2-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row justify-content-center"
    >
      <div
        className="col"
      >
        <button
          className="btn btn-game btn-lg btn-danger"
          onClick={[Function]}
          type="button"
        >
          Reset
        </button>
      </div>
      <div
        className="col"
      >
        <a
          className="btn btn-game btn-lg btn-danger"
          href="/"
          onClick={[Function]}
        >
          Back
        </a>
      </div>
    </div>
  </div>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "bottom": 16,
        "left": 16,
        "pointerEvents": "none",
        "position": "fixed",
        "right": 16,
        "top": 16,
        "zIndex": 9999,
      }
    }
  />
</div>
`;
